src/Scripting/Core/ScriptOptions.cs

```diff
@@ -19,11 +21,51 @@ public sealed class ScriptOptions
     {
         public static ScriptOptions Default { get; } = new ScriptOptions(
             filePath: "",
-            references: ImmutableArray<MetadataReference>.Empty,
+            references: GetDefaultMetadataReferences(),
             namespaces: ImmutableArray<string>.Empty,
             metadataResolver: RuntimeMetadataReferenceResolver.Default,
             sourceResolver: SourceFileResolver.Default);
 
+        private static ImmutableArray<MetadataReference> GetDefaultMetadataReferences()
+        {
+            string corlibDir = RuntimeMetadataReferenceResolver.DefaultCorLibDirectoryOpt;
+
+            if (corlibDir == null)
+            {
+                return ImmutableArray<MetadataReference>.Empty;
+            }
+
+            // Provide similar surface to mscorlib (netstandard 2.0).
+            // These references are resolved lazily. Keep in sync with list in core csi.rsp.
+            var files = new[]
```

Comment:
So my goal was to emulate more or less mscorlib. On desktop you need to ```#r "System"``` and ```#r "System.Core"``` if you want anything outside of mscorlib, such as LINQ and XML. So it makes sense that you'd need to ```#r "System.Xml.XDocument"```. Note that ```#r``` is recursive (like a nuget reference), so it adds a reference to specified assembly and all its dependencies. Since System.Xml.XDocument depends on System.Xml.ReaderWriter you just need to ```#r``` the former.

Good catch on System.Reflection.Extensions - I'll add that one.